server.js

```diff
@@ -1,36 +1,282 @@
+      releases: artistQuery
+    });
+  }
+});
+
+// This route will return a collection of releases with a title containing the specified word(s) 
+app.get("/releases/title/:title", async (req, res) => {
+  const titleQuery = await Release.find({
+    name: new RegExp(req.params.title, "i")
+  }).populate("album_type").populate("artists")
+  if (titleQuery.length === 0) {
+    res.status(404).json(ERROR_RELEASES_NOT_FOUND);
+  } else {
+    res.json({
+      total: titleQuery.length,
+      releases: titleQuery
+    });
+  }
+});
+
+// This route will return a collection of releases with a type containing the specified word(s)
+app.get("/releases/type/:type", async (req, res) => {
+  const typeQuery = await Release.aggregate([
+    {
+      $lookup: {
+        from: "types",
+        localField: "album_type",
+        foreignField: "_id",
+        as: "album_type"
+      }
+    },
+    {
+      $lookup: {
+        from: "artists",
+        localField: "artists",
+        foreignField: "_id",
+        as: "artists"
+      }
+    },
+    {
+      $match: {
+        "album_type.album_type": new RegExp(req.params.type, "i")
+      }
+    }
+  ]);
+  if (typeQuery.length === 0) {
+    res.status(404).json(ERROR_RELEASES_NOT_FOUND);
+  } else {
+    res.json({
+      total: typeQuery.length,
+      releases: typeQuery
+    });
+  }
+});
+
+// This route will return a collection of releases in the specified market for the specified type
+app.get("/releases/market/:market/type/:type", async (req, res) => {
+  const filteredReleases = await Release.aggregate([
+    {
+      $lookup: {
+        from: "types",
+        localField: "album_type",
+        foreignField: "_id",
+        as: "album_type"
+      }
+    },
+    {
+      $lookup: {
+        from: "artists",
+        localField: "artists",
+        foreignField: "_id",
+        as: "artists"
+      }
+    },
+    {
+      $match: {
+        $and: [
+          { "album_type.album_type": new RegExp(req.params.type, "i") },
+          { available_markets: req.params.market }
+        ]
+      }
+    }
+  ]);
+  if (filteredReleases.length === 0) {
+    res.status(404).json(ERROR_RELEASES_NOT_FOUND);
+  } else {
+    res.json({
+      total: filteredReleases.length,
+      releases: filteredReleases
+    });
+  };
+});
+
+// This route will return a list of all available release types.
+app.get("/types", async (req, res) => {
+  const allTypes = await Type.find();
+  if (allTypes.length === 0) {
+    res.status(404).json(ERROR_RELEASES_NOT_FOUND);
+  } else {
+    res.json({
+      total: allTypes.length,
+      types: allTypes
+    });
+  }
+});
+
+// This route will return a list of all available artists.
+app.get("/artists", async (req, res) => {
+  const allArtists = await Artist.find();
+  if (allArtists.length === 0) {
+    res.status(404).json(ERROR_RELEASES_NOT_FOUND);
+  } else {
+    res.json({
+      total: allArtists.length,
+      artists: allArtists
+    });
+  }
+});
```

Comment:
Once again, i am super impressed by your project this week. I wish i could give you more feedback, but your code is in a level where i am not at the moment. I hope maybe a code coach can give you some more comments!